### PR TITLE
Create service skeleton with core application (GSI-121)

### PR DESCRIPTION
**Pull request by Cito**
This PR adapts the service template to the needs of the access request service and adds a skeleton for the implementation with a simple repository for access requests as core application.

### CODE REVIEW

**Comment by Cito** [comment]
This has now been superseded by #2.

File: ars/core/repository.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            request = await self._dao.get_by_id(access_request_id)
+        except ResourceNotFoundError as error:
+            raise self.AccessRequestNotFoundError("Access request not found") from error
+        if request.status == status:
+            raise self.AccessRequestError("Same status is already set")
+        if request.status != AccessRequestStatus.PENDING:
+            raise self.AccessRequestError("Status cannot be reverted")
+
+        # Should set the status in the claims repository here
+        # and proceed only if this succeeds.
```
Comment (KerstenBreuer) [inline ars/core/repository.py:177]: We submit a claim if approved and do nothing if rejected, right?

**Review by KerstenBreuer — APPROVED** [review]
Looks good just some minor structural suggestions.

File: ars/models.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from pydantic import BaseModel, Field
+
+
+class MessageBase(BaseModel):
+    """A message base container"""
+
+    message: str = Field(..., description="The message content.")
+    created_at: datetime = Field(
+        ..., description="The date/time when the message was created"
+    )
```
[the remark this answers is not in the document]
Comment (Cito) [reply ars/models.py:29]: Remnant from the template. Has been removed in #2.

File: ars/core/repository.py (hunk 1 above)
Comment (Cito) [reply ars/core/repository.py:177]: Exactly.

File: ars/adapters/outbound/dao.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    @staticmethod
+    async def construct(
+        *, config: AccessRequestConfig, dao_factory: DaoFactoryProtocol
+    ) -> AccessRequestDaoPort:
+        """Setup the DAOs using the specified provider of the
+        DaoFactoryProtocol."""
+
+        return await dao_factory.get_dao(
+            name=config.access_requests_collection,
```
[the remark this answers is not in the document]
Comment (Cito) [reply ars/adapters/outbound/dao.py:57]: Will change it in #2. Yes, I didn't like the reuse either.

File: ars/adapters/inbound/http/__init__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
```
[the remark this answers is not in the document]
Comment (Cito) [reply ars/adapters/inbound/http/__init__.py:1]: Will change that in #2.

File: ars/core/repository.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+DATA_STEWARD_ROLE = "data_steward"
+
+
+class AccessRequestConfig(BaseSettings):
+    """Config parameters needed for the AccessRequestRepository."""
+
+    access_requests_collection: str = Field(
+        ...,
+        description="The name of the database collection for access requests",
+    )
```
[the remark this answers is not in the document]
Comment (Cito) [reply ars/core/repository.py:47]: Makes sense. Will change that in #2.

**Pull request merged**